File: problems/python/001/multiples.py
"""
If we list all the natural numbers below 10 that are multiples of 3 or 5, we
get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
"""

if __name__ == '__main__':
    multiples_sum = 0
    for i in range(1, 1000):
        # Check if i is multiple of 3 or 5
        if (i % 3 == 0) or (i % 5 == 0):
            multiples_sum += i

    print(multiples_sum)

File: problems/ruby/001/multiples.rb
### Jaime Alvarez -- August 2016

=begin
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
=end

puts "Sum of all multiples of 3 or 5 below 1000 = "
puts (1..1000).to_a.select { |element| element % 3 == 0 || element % 5 == 0 }.inject(:+)
